src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/PackageRestoreDataSource.cs

```diff
@@ -135,6 +145,17 @@ private async Task<bool> RestoreCoreAsync(PackageRestoreUnconfiguredInput value)
 
                 _latestHash = hash;
 
+                if (_cycleDetector.ComputeCycleDetection(hash))
```

Comment:
Even if we hit this here, still nominate on line 160 right?

That should probably not happen correct?

Reply:
I think we should return succeed = false.